test/verifier/customSmartWalletDeployVerifier.test.ts

```diff
@@ -1,205 +1,319 @@
+        [fakeToken.address]: true
+      })
+      const result = deployVerifierMock.acceptToken(fakeToken.address);
+      await expect(result).to.be.revertedWith('Token is already accepted');
+    });
+
+    it('should revert if accepting a token with ZERO ADDRESS', async function () {
+      const result = deployVerifierMock.acceptToken(constants.AddressZero);
+
+      await expect(result).to.be
+        .revertedWith('Token cannot be zero address');
+    });
+
+    it('should revert if caller is not the owner', async function () {
+      const [, other] = await ethers.getSigners();
+
+      await expect(
+        deployVerifierMock.connect(other).acceptToken(fakeToken.address)
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('getAcceptedTokens()', function () {
+    it('should get all the accepted tokens', async function () {
+      const fakeTokenList = [fakeToken.address];
+      await deployVerifierMock.setVariable('acceptedTokens', fakeTokenList);
+
+      const acceptedTokens = await deployVerifierMock.getAcceptedTokens();
+      expect(acceptedTokens).to.deep.equal(fakeTokenList);
+    });
+  });
+
+  describe('acceptsToken()', function () {
+    beforeEach(async function(){
+      const { address } = fakeToken;
+      await deployVerifierMock.setVariable('tokens', {
+          [address]: true
+      })
+    })
+
+    it('should return true if token is accepted', async function () {
+      const acceptsToken = await deployVerifierMock.acceptsToken(fakeToken.address);
+      expect(acceptsToken).to.be.true;
+    });
+
+    it('should return false if token is not accepted', async function () {
+      const fakeTokenUnaccepted = await smock.fake<ERC20>('ERC20');
+      const acceptsToken = await deployVerifierMock.acceptsToken(fakeTokenUnaccepted.address);
+      expect(acceptsToken).to.be.false;
+    });
+  });
+
+  describe('versionVerifier()', function () {
+    it('should get the current version', async function () {
+      const version = await deployVerifierMock.versionVerifier();
+      expect(version).to.eq('rif.enveloping.token.iverifier@2.0.1');
+    });
+  });
+
+  describe.skip('_isContract()', function () {
+    //Currently it's not possible to test internal functions. 
+    // It might be in the future: https://github.com/defi-wonderland/smock/issues/106
+    it('should return true if a contract has code in it', function () {
+        console.log('Currently unmockable with smock')
+    });
+  });
+
+  describe('verifyRelayedCall()', function () {
+    let owner: SignerWithAddress;
+    let recipient: SignerWithAddress;
+    let relayWorker: SignerWithAddress;
+    let fakeRelayHub: FakeContract<RelayHub>;
+
+    beforeEach(async function() {
+      [owner, recipient, relayWorker] = await ethers.getSigners();
+      fakeRelayHub = await smock.fake<RelayHub>('RelayHub');
+    })
+
+    it('should not revert', async function () {
```

Comment:
I am assuming that this is the happy path?

Reply:
Yes this is the happy path